### PR TITLE
fix: serialize booleans as native JSON in precomputed request body

### PR DESCRIPTION
**Pull request by leoromanovsky**
## Motivation

The `buildRequestBody()` method in `EppoPrecomputedClient` was converting all non-numeric attribute values to strings, including booleans. This caused boolean attributes to be sent in the wrong wire format:

**Before (incorrect):**
```json
{
  "categoricalAttributes": {
    "isPremium": "true",
    "hasPushEnabled": "false"
  }
}
```

**After (correct):**
```json
{
  "categoricalAttributes": {
    "isPremium": true,
    "hasPushEnabled": false
  }
}
```

This fix aligns the Android SDK with the JS SDK and iOS SDK (see [ios-sdk#95](https://github.com/Eppo-exp/eppo-ios-sdk/pull/95)), which preserve native boolean types in the wire format.

The shared test data in `sdk-test-data/configuration-wire/precomputed-v1.json` confirms that booleans should be native JSON booleans:
```json
"categoricalAttributes": {
  "hasPushEnabled": false,
  "buildNumber": 42
}
```

## Changes

- Add `ContextAttributesSerializer` utility class for consistent attribute serialization
- Properly handle booleans as native JSON booleans in categorical attributes
- Exclude null values from serialization (matching JS/iOS behavior)
- Simplify `buildRequestBody()` by using the new utility
- Add comprehensive unit tests for wire format verification
- Add `precomputed-v1.json` test data for integration tests

## Decisions

- Created a dedicated utility class (`ContextAttributesSerializer`) for testability and reuse across subject attributes and bandit actions
- Used `Map<String, Object>` for categorical attributes instead of `Map<String, String>` to allow native boolean values

## Test plan

- [x] Run unit tests: `./gradlew :eppo:testDebugUnitTest`
- [x] Run instrumented tests: `./gradlew :eppo:connectedDebugAndroidTest`
- [x] Verify wire format matches expected output in real API calls

### CODE REVIEW

File: eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+   */
+  public static Map<String, Object> serialize(Attributes attributes) {
+    Map<String, Object> result = new HashMap<>();
+    Map<String, Number> numericAttrs = new HashMap<>();
+    Map<String, Object> categoricalAttrs = new HashMap<>();
+
+    if (attributes != null) {
+      for (String key : attributes.keySet()) {
+        EppoValue value = attributes.get(key);
+        if (value != null && !value.isNull()) {
```
Comment (aarsilv) [inline eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:42]: Should we be adding `null` as the value if it _is_ null?

File: eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          if (value.isNumeric()) {
```
Comment (aarsilv) [inline eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:43]: I wonder if the newly-added `unrwap()` could help simplify things here. 

https://github.com/Eppo-exp/sdk-common-jdk/blob/main/src/main/java/cloud/eppo/api/EppoValue.java#L110

File: eppo/src/test/java/cloud/eppo/android/ContextAttributesSerializerTest.java [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    // Verify booleans are stored as Boolean objects, not Strings
+    assertTrue(categoricalAttrs.get("isPremium") instanceof Boolean);
+    assertTrue(categoricalAttrs.get("hasNotifications") instanceof Boolean);
+
+    assertEquals(true, categoricalAttrs.get("isPremium"));
+    assertEquals(false, categoricalAttrs.get("hasNotifications"));
+  }
+
+  @Test
+  public void testBooleansNotConvertedToStrings() {
```
Comment (aarsilv) [inline eppo/src/test/java/cloud/eppo/android/ContextAttributesSerializerTest.java:95]: 💪

File: eppo/src/test/java/cloud/eppo/android/ContextAttributesSerializerTest.java [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+import org.robolectric.RobolectricTestRunner;
+
+/**
+ * Tests for ContextAttributesSerializer wire format.
+ *
+ * <p>These tests ensure that attributes are serialized correctly to match the wire format expected
+ * by the precomputed flags API and to maintain consistency with other Eppo SDKs (JS, iOS).
+ */
+@RunWith(RobolectricTestRunner.class)
+public class ContextAttributesSerializerTest {
```
Comment (aarsilv) [inline eppo/src/test/java/cloud/eppo/android/ContextAttributesSerializerTest.java:24]: I like that we pulled this out into it's own testable class. Since we're doing that, let's include a test for `null` valued attributes to cover that base.

**Review by aarsilv — APPROVED** [review]
Good catch and glad we have a shared test case now that exercises this.

Approving with two minor comments you can take or leave:
* The new [`EppoValue.unwrap()`](https://github.com/Eppo-exp/sdk-common-jdk/blob/main/src/main/java/cloud/eppo/api/EppoValue.java#L110) method may be useful
* Consideration of `null`-valued attributes

File: eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java (hunk 2 above)
Comment (leoromanovsky) [reply eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:43]: Thanks, good suggestion, hadn't explore the common java repo.

File: eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java (hunk 1 above)
Comment (leoromanovsky) [reply eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:42]: We explicitly skip null in swift and javascript so I was matching that behavior.

File: eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java (hunk 2 above)
Comment (leoromanovsky) [reply eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:43]: It doesn't simplify the code for us because we need to know the variation type, would look like this:

```
  // We'd still need the same if/else chain to pick the VariationType
  if (value.isNumeric()) {
      numericAttrs.put(key, value.unwrap(VariationType.NUMERIC));   // Double
  } else if (value.isBoolean()) {
      categoricalAttrs.put(key, value.unwrap(VariationType.BOOLEAN)); // Boolean
  } else {
      categoricalAttrs.put(key, value.unwrap(VariationType.STRING));  // String
  }
```

Is that what you were trying to simplify?

Comment (leoromanovsky) [reply eppo/src/main/java/cloud/eppo/android/util/ContextAttributesSerializer.java:43]: It would be more helpful to augment the `EppoValue` in jdk common to auto detect the type:

```
public Object unwrap() {
  switch (this.type) {
    case BOOLEAN: return boolValue;
    case NUMBER: return doubleValue;
    case STRING: return stringValue;
    ...
  }
}
```

Then the serializer becomes:

```
if (value.isNumeric()) {
  numericAttrs.put(key, (Number) value.unwrap());
} else {
  categoricalAttrs.put(key, value.unwrap());
}
```
